### DISCUSSION TITLE
Macroscale RF EM simulations (like HFSS/CST/ADS)

### DISCUSSION THREAD

**Discussion by prwang** [flexcompute/tidy3d#1485] [Ideas]
I wonder if we can replace those bulky commercial RF EDA software with this novel FDTD solver. If this could be successful, the impact will expand from the academia researchers to a larger RF engineer community. Here are the proposed steps, say, if we want to model an printed circuit board antenna in tidy3d:
1. Material. We need to prepare the macroscale behavior of Copper and FR4 dielectric. For the dielectric, we have manufacturer's datasheet of \eps and \delta values as a function of frequency (e.g. https://www.midwestpcb.com/data_sheets/IsolaFR-408.pdf) , we may  fit a curve using tidy3d.Drude.  Copper: dispersionless tidy3d.Medium(conductivity =5.8e1) 
2. Modeling and geometry.
3. Excitation. Mode source, inject TEM mode (lowest) in the microstrip lines.
4. Boundary. PML. (Or BlochBoundary, for phased arrays)
5. Simulation: Radically tune the spatial and time step to match the wavelength of centimeters and time step of several nanoseconds! (How?)
6. Post processing. FieldProjectionAngleMonitor, s11(f), radiation pattern (f,\theta,\phi, polarization), phase center.

**Comment by weiliangjin2021** [comment]
Hi @prwang, thank you for sharing your proposal. It's a nice idea, and we are quite interested in incorporating such a capability! Do you have any specific application examples in mind so that we can work through the list of all missing features?

> 1. Material. We need to prepare the macroscale behavior of Copper and FR4 dielectric. For the dielectric, we have manufacturer's datasheet of \eps and \delta values as a function of frequency (e.g. https://www.midwestpcb.com/data_sheets/IsolaFR-408.pdf) , we may  fit a curve using tidy3d.Drude.  Copper: dispersionless tidy3d.Medium(conductivity =5.8e1)

Aside from Drude, we support a more generic dispersion model, and you can find the fitting tool [here](https://docs.flexcompute.com/projects/tidy3d/en/latest/notebooks/Fitting.html).

> 2. Modeling and geometry.

In your application, what's the most convenient way in building the geometry? Do you build directly from elementary geometries such as polyslabs, cylinders, etc., or import from files such as CAD? 

> 3. Excitation. Mode source, inject TEM mode (lowest) in the microstrip lines.

We have mode source that computes the modal field profile and effective index. I guess for RF, it'll be more convenient if there are convenient functions that computes impedance, beta, and gamma? 

> 4. Boundary. PML. (Or BlochBoundary, for phased arrays)

Both are supported already.

> 5. Simulation: Radically tune the spatial and time step to match the wavelength of centimeters and time step of several nanoseconds! (How?)

You mean when the wavelength is cm scale, the time steps will be too many because one wants to simulate the time evolution over ~ns?

> 6. Post processing. FieldProjectionAngleMonitor, s11(f), radiation pattern (f,\theta,\phi, polarization), phase center.

Those are all good to add.
